pkg/kv/kvserver/node_liveness.go

```diff
@@ -424,35 +436,69 @@ type livenessUpdate struct {
 	oldRaw []byte
 }
 
-func (nl *NodeLiveness) setDecommissioningInternal(
-	ctx context.Context, nodeID roachpb.NodeID, liveness LivenessRecord, decommission bool,
-) (changeCommitted bool, err error) {
-	update := livenessUpdate{
-		newLiveness: kvserverpb.Liveness{
+func (nl *NodeLiveness) setMembershipStatusInternal(
+	ctx context.Context,
+	nodeID roachpb.NodeID,
+	oldLivenessRec LivenessRecord,
+	targetStatus kvserverpb.MembershipStatus,
+) (statusChanged bool, err error) {
+	// Lets compute what our new liveness record should be.
+	var newLiveness kvserverpb.Liveness
+	if oldLivenessRec.Liveness == (kvserverpb.Liveness{}) {
+		// Liveness record didn't previously exist, so we create one.
+		//
+		// ASK(reviewers): Why is it possible for a liveness record to not exist
```

Comment:
It is just generally difficult to write it at startup. When a node joins the cluster, this completes before it has had a chance to write its liveness record. If it gets decommissioned immediately, there won't be one yet. The Connect RPC can solve this though, I think? We can bootstrap clusters with a liveness record for n1. Any other node at some point has to join the cluster for the first time via the Connect RPC, which as part of its job can make sure the liveness record exists before responding to the new node.